### PR TITLE
Decouple system timer and drivers: SoC Initialization

### PR DESCRIPTION
**Pull request by pizi-nordic**
During our work in timer area, we found that system clock frequency is used incorrectly in several places.
For example, it is used to obtain CPU frequency needed for SoC initialization.
If we select different system timer device, all these calculations become invalid.

This PR fixes the problem by obtaining all needed information directly from the DTS.
Brings us closer to: #15363

### CODE REVIEW

File: ext/hal/ti/simplelink/source/ti/devices/msp432p4xx/startup_system_files/system_msp432p401m.c [hunk 1]
```diff
@@ -44,6 +44,7 @@
 
 #include <stdint.h>
 #include <ti/devices/msp432p4xx/inc/msp.h>
+#include <generated_dts_board.h>
```
Comment (ioannisg) [inline ext/hal/ti/simplelink/source/ti/devices/msp432p4xx/startup_system_files/system_msp432p401m.c]: I am not so happy with including generated headers in ext.
Could this be done in soc.h, instead?

Comment (ioannisg) [reply ext/hal/ti/simplelink/source/ti/devices/msp432p4xx/startup_system_files/system_msp432p401m.c]: that's what we do for nRF

**Review by ioannisg — APPROVED** [review]
Did a pass - it looks ok to me. I left a comment about including auto-generated headers.
The PR does not include changed to nRF platform. Needs review from more collaborators.

**Review by tyrantlizard77 — COMMENTED** [review]
question I have...we are now setting the CPU clock in terms of DT_CPUS_CPU_0_CLOCK_FREQUENCY, but does CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC also derive from this value or is it still manually set in defconfig?

**Comment by pizi-nordic** [comment]
@andrewboie: The CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC  is still set manually in defconfig. However all these changes are preparation for removal of of this define. I noticed, that it is widely used in wrong context as well as in many places we are assuming that system clock frequency is known at compile time which is not always true.

**Review by anangl — APPROVED** [review]

**Pull request closed without merging**